src/Mpi/MpiControl.cpp

```diff
@@ -589,35 +590,118 @@ void MpiControlType<ndim, ParticleType>::UpdateMpiGhostParents
 
   const int total_ghost_receive = std::accumulate(N_updates_from_proc.begin(),
                                                   N_updates_from_proc.end(), 0);
-  vector<FLOAT> buffer_receive(total_ghost_receive);
-  vector<int> buffer_receive_iorig(total_ghost_receive);
 
-  // Send the updated masses and the iorigs (to retrieve the parent particles)
-  MPI_Alltoallv(&buffer[0], &N_updates_per_proc[0], &displs_ghosts[0], GANDALF_MPI_FLOAT,
-                &buffer_receive[0], &N_updates_from_proc[0], &displs_ghosts_receive[0],
-                GANDALF_MPI_FLOAT, MPI_COMM_WORLD);
+  vector<ReturnDataType> buffer_receive(total_ghost_receive);
+
+  MPI_Datatype MpiReturnType;
+  MPI_Datatype types[1] = {MPI_BYTE};
+  MPI_Aint offsets[1] = {0};
+  int blocklen[1] = {sizeof(ReturnDataType)};
 
-  MPI_Alltoallv(&buffer_iorig[0], &N_updates_per_proc[0], &displs_ghosts[0], MPI_INT,
-                &buffer_receive_iorig[0], &N_updates_from_proc[0], &displs_ghosts_receive[0],
-                MPI_INT, MPI_COMM_WORLD);
+  MPI_Type_create_struct(1,blocklen,offsets,types,&MpiReturnType);
```

Comment:
I am not sure we want to create a MPI data type every time we call this function